pkg/services/ngalert/models/alert_rule.go

```diff
@@ -162,6 +162,11 @@ type AlertRule struct {
 	IsPaused    bool
 }
 
+type AlertRuleWithOptionals struct {
```

Comment:
This is something I proposed to avoid having to pass in additional arguments. It is meant to be a stop-gap measure until we can pass request structs. For example `AlertRuleUpdateRequest` that let's us encapsulate the changes that we want to apply, rather than the end state of the alert rule, which is not possible for partial updates.